api/v3/Pcpteams.php

```diff
@@ -1101,7 +1101,8 @@ function _civicrm_api3_pcpteams_getMoreInfo(&$params) {
     if($entityFile){
       $fileInfo = reset($entityFile);
       $fileId   = $fileInfo['fileID'];
-      $imageUrl = CRM_Utils_System::url('civicrm/file',"reset=1&id=$fileId&eid={$pcpId}"); 
+      $fileHash = CRM_Core_BAO_File::generateFileHash($pcpId, $fileId);
+      $imageUrl = CRM_Utils_System::url('civicrm/file',"reset=1&id=$fileId&eid={$pcpId}&fcs=$fileHash"); 
```

Comment:
This looks about right, with some provisos.

[The advisory](https://civicrm.org/advisory/civi-sa-2019-01-weak-access-control-for-file-attachments) recommends using the `Attachment.get` API to be backwards+forwards compatible. For example:

```
[bknix-dfl:~/bknix/build/dmaster/sites/all/modules/civicrm]$ cv api attachment.getvalue id=2 return=url
"http://dmaster.bknix:8001/civicrm/file?reset=1&id=2&eid=625&fcs=aaaaaaaaa8a413bdf5d8f4424bcf85cccb26f18cf4ff983f3cf75be31f8529ad_1550866111_111"
```

Backwards+forwards means the same line of code should yield a valid URL on past, present, and future releases.

However, the advice might not work here -- because the current version of `Attachment.get` will complain if it sees an unfamiliar `entity_table` (and I'm guessing this extension attaches to something... not on the whitelist). That's supposed to be relaxed for PHP-based consumers... in an [upcoming version](https://lab.civicrm.org/dev/core/issues/690)... but it won't help here and now...

So I think the only option is calling `generateFileHash()` -- like this patch does. You need something that works!

One suggestion:

```php
// Current
$fileHash = CRM_Core_BAO_File::generateFileHash($pcpId, $fileId);

// Suggested
$fileHash = is_callable(['CRM_Core_BAO_File', 'generateFileHash']) 
  ? CRM_Core_BAO_File::generateFileHash($pcpId, $fileId)
  : '';
```

That's not quite as maintainable/compatible as using the API, but it'll be a little better (e.g. it won't crash when deployed on an older version).

Reply:
Yeah, good point about being compatible with old insecure versions of CiviCRM. Not that we really want to enable such bad behaviour ...

Reply:
lol. Well... yeah...